components/butterfly/tests/common/mod.rs

```diff
@@ -179,17 +180,33 @@ impl SwimNet {
         from.remove_from_block_list(to.member_id());
     }
 
-    #[allow(clippy::assertions_on_constants)]
-    #[allow(clippy::match_wild_err_arm)]
     pub fn health_of(&self, from_entry: usize, to_entry: usize) -> Option<Health> {
+        /// To avoid deadlocking in a test, we use `health_of_by_id_with_timeout` rather than
+        /// `health_of_by_id`.
+        const HEALTH_OF_TIMEOUT: Duration = Duration::from_secs(5);
+
         let from = self.members
                        .get(from_entry)
                        .expect("Asked for a network member who is out of bounds");
 
         let to = self.members
                      .get(to_entry)
                      .expect("Asked for a network member who is out of bounds");
-        from.member_list.health_of_by_id(to.member_id())
+
+        match from.member_list
+                  .health_of_by_id_with_timeout(to.member_id(), HEALTH_OF_TIMEOUT)
+        {
+            Ok(health) => Some(health),
+            Err(Error::UnknownMember(_)) => None,
+            Err(Error::Timeout(_)) => {
+                panic!("Timed out after waiting {:?} querying member health",
```

Comment:
do we want to panic here?

Reply:
We do indeed. This function is only used by tests and if we've timed out it implies we're deadlocked, so we should fail the test. Panicking is the most direct way to fail; it's equivalent to `assert!(false, …)` but won't run afoul of https://rust-lang.github.io/rust-clippy/master/#assertions_on_constants.